### PR TITLE
Refs [TASK][26165]: Create View Presenter contract FragmentDetail

### PR DESCRIPTION
**Pull request by aug16vcc**
## Related Tickets
- [#Ticket ID](https://edu-redmine.sun-asterisk.vn/issues/26165)

## WHAT
- Create View Presenter contract FragmentDetail

## HOW
- Create DetailContract, implement view and presenter, fix repository interface

## WHY
- needed task

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Vu Chi Cong | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/SummaryWeatherAdapter.kt [hunk 1]
```diff
@@ -41,6 +43,13 @@ class SummaryWeatherAdapter(
                 textTemperature.text = unitSystem.formatTemperature(item.temperature, resources)
                 textDateTime.text = TimeUtils.formatToString(timeFormatString, item.dt)
                 containerItem.setOnClickListener { itemClickListener(item) }
+                item.icon?.let {
+                    Glide.with(context)
+                        .load(Uri.parse(context.getString(R.string.format_url_weather_icon, it)))
+                        .centerCrop()
+                        .error(R.drawable.ic_round_broken_image_24)
+                        .into(imageWeather)
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/SummaryWeatherAdapter.kt]: Viết extension cho ImageView luôn 
fun ImageView.load(...)

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/SummaryWeatherAdapter.kt [hunk 2]
```diff
@@ -41,6 +43,13 @@ class SummaryWeatherAdapter(
                 textTemperature.text = unitSystem.formatTemperature(item.temperature, resources)
                 textDateTime.text = TimeUtils.formatToString(timeFormatString, item.dt)
                 containerItem.setOnClickListener { itemClickListener(item) }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/SummaryWeatherAdapter.kt]: mỗi lần bind data nó phải setClick lại sao

File: app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailContract.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import com.sunasterisk.boringweather.data.model.DetailWeather
+
+interface DetailContract {
+    interface View {
+        val presenter: Presenter?
+
+        fun showError(@StringRes errorStringRes: Int)
+        fun showDetailWeather(detailWeather: DetailWeather)
+        fun finishRefresh()
+        fun loadCity(city: City)
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailContract.kt]: sao View lại loadCity nhỉ?

**Review by datdt-1935 — APPROVED** [review]

File: app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailFragment.kt [hunk 4]
```diff
@@ -93,14 +121,32 @@ class DetailFragment : BaseFragment() {
...
     companion object {
         private const val ARGUMENT_CITY_ID = "city_id"
         private const val ARGUMENT_DAILY_WEATHER_DATE_TIME = "dailyWeather_dt"
+        private const val ARGUMENT_FOCUS_HOURLY_WEATHER_DATE_TIME = "focus_hourlyWeather_dt"
 
-        fun newInstance(cityId: Int, dailyWeatherDateTime: Long) = DetailFragment().apply {
+        fun newInstance(
+            cityId: Int,
+            dailyWeatherDateTime: Long,
+            focusHourlyWeatherDateTime: Long? = null
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailFragment.kt:147]: Nhiều thuộc tính thì có thể gộp vào 1 model rồi gửi nó qua được không

File: app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailContract.kt (hunk 3 above)
Comment (huytq-2422) [reply app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailContract.kt]: Sao không lưu ở presenter luôn đỡ phải chuyển lên View

Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/ui/detail/DetailContract.kt]: Sorry đặt t sai tên

**Pull request merged**